finetuner/experiment.py

```diff
@@ -139,6 +148,48 @@ def create_run(
             if isinstance(callback, EvaluationCallback):
                 eval_callback = callback
 
+        if isinstance(train_data, (TextIO)) or (
```

Comment:
please move all this logic to the `from_csv` function. You can rename the function `build_dataset` and put it under a `dataset.py` file. All the dataset resolution logic should go there